internal/api/mfa.go

```diff
@@ -132,13 +132,16 @@ func (a *API) EnrollFactor(w http.ResponseWriter, r *http.Request) error {
 	}
 	svgData.End()
 
-	factor, err := models.NewFactor(user, params.FriendlyName, params.FactorType, models.FactorStateUnverified, key.Secret())
-	if err != nil {
-		return internalServerError("database error creating factor").WithInternalError(err)
-	}
+	factor := models.NewFactor(user, params.FriendlyName, params.FactorType, models.FactorStateUnverified, key.Secret())
+
 	err = a.db.Transaction(func(tx *storage.Connection) error {
 		if terr := tx.Create(factor); terr != nil {
+			pgErr := utilities.NewPostgresError(terr)
+			if pgErr.IsUniqueConstraintViolated("mfa_factors_user_friendly_name_unique") {
+				return internalServerError(fmt.Sprintf("a factor with the friendly name %q for this user already exists", factor.FriendlyName))
```

Comment:
In general I'd prefer if we just checked for _any_ unique constraint violation error instead of a specific one. It can be very easy to forget about this and rename the constraint in the future, which would brake this.